### ISSUE TITLE
Triggering css file reload (breaks browersync)

### ISSUE THREAD

**Issue by razor-x** [bmatcuk/html-brunch-static#6]
Made a minimal example to reproduce this issue: https://github.com/rxbugs/static-html-brunch-browsersync.

Not sure if this is a bug here or with sass-brunch: see the workaround below.

## Issue

This is triggering changes to `app.css`
when any of the `.static` files are changed.
Somehow this breaks browsersync as it just injects the new css file,
but does not trigger a page reload.

## Reproduction steps

```
$ git clone git@github.com:rxbugs/static-html-brunch-browsersync.git
$ cd static-html-brunch-browsersync
$ npm install
$ npm start
```

Now try to make changes to `app/styles/main.scss` and `app/index.static.hbs`
or `app/layouts/main.static.hbs` and notice how manual reload is needed to
see html changes.

Note that if you remove the brunch config line

```
stylesheets: joinTo: 'app.css'
```

then this disables styles generation and restores automatic reload of html
changes.

Note that the commit (61aa0c4) only has `app/assets/index.html` and not
`static-html-brunch`. That case shows the normal expected behavior.

## Workaround

Using

```
stylesheets: joinTo: 'app.css' : /^app\/styles/
```

to only watch the `styles` path seems to fix the problem.

## Environment

```
$ npm version
{ 'simple-brunch': '0.1.0',
  npm: '3.3.12',
  ares: '1.10.1-DEV',
  http_parser: '2.6.0',
  icu: '56.1',
  modules: '47',
  node: '5.5.0',
  openssl: '1.0.2e',
  uv: '1.8.0',
  v8: '4.6.85.31',
  zlib: '1.2.8' }
```

**Comment by bmatcuk** [comment]
Hmm, this is a very strange issue. My guess is that it has something to do with the way the brunch browsersync plugin is determining what files have changed. The brunch static plugin (on which the html brunch static plugin is dependent) is kind of a hack since brunch doesn't have a native way to output arbitrary files. The solution you found (configuring brunch to watch the app/styles directory for css files instead of app) may be the only way to get it working =(

I'll look into it in more detail sometime this week when time allows. Thanks for the detailed bug report!

**Comment by bmatcuk** [comment]
After reading through brunch's code and browser-sync-brunch's code, I'm pretty sure this problem can't be solved =( Here's why:

* Brunch has no support for creating arbitrary files. It only knows about the javascript and stylesheet files that it creates. When any file is changed that affects one of the javascript or stylesheet outputs, the javascript/stylesheet file is added to a list of "changed" files.
  * Brunch plugins must register themselves as affecting either the javascript file or the stylesheet file. [brunch-static](http://github.com/bmatcuk/brunch-static) registers itself as affecting stylesheets. This is because it doesn't _actually_ write anything to the output. If it wrote to the javascript file, brunch would automatically add the commonjs wrapper around the "nothing" that brunch-static outputs, adding unnecessary length to the javascript file. On the other hand, brunch doesn't automatically add anything to the stylesheet file, so writing nothing literally writes nothing. Had brunch-static chosen javascript files instead, however, you'd still have the same issue except that the javascript file would get updated when making changes to static files, instead of the stylesheet. Plugins have no way of adding files to the list of changed files.
  * Brunch also understands static assets such as images or static html files. If any of them change, brunch copies them, verbatim, to the output directory. Updated assets are _not_ added to the list of changed files.
* browser-sync-brunch receives the list of changed files. As mentioned above, this only includes the javascript and stylesheets that brunch is creating.
  * If the list of changed files is empty, this indicates that some static asset must have changed to trigger the update. browser-sync-brunch instructs browser-sync to reload the page.
  * If the list is not empty, browser-sync-brunch sends the list to browser-sync, which then grabs _only_ the file(s) that changed and "hot reloads" them.

This explains the behavior you're seeing. Changing one of the handlebars files is "changing" the stylesheet, instructing browser-sync to reload the stylesheet so you don't see any changes. But if you setup brunch to only combine files from `app/styles` into the stylesheet, then changing one of the handlebars files confuses brunch. It causes a compile, but doesn't update either of the javascript or stylesheet files made by brunch, so the list of changed files is empty. browser-sync-brunch sees this as an asset update and forces the page to reload.

tl;dr: I can't fix this problem, but your proposed workaround works nicely. In fact, restricting the stylesheet files in any way that will exclude the files that html-brunch-static processes will work... for example, you could do:

```javascript
stylesheets: joinTo: 'app.css' : /\.css$/
```

or:

```javascript
stylesheets: joinTo: 'app.css' : /^((?!\.static\.).)*$
```

(see [here](http://stackoverflow.com/a/406408/2836512) for an explanation of that regex - it basically matches any file that does _not_ have `.static.`).

Thanks for the detailed bug report! I'll leave this open for other people to find... maybe brunch will eventually support arbitrary files so I can fix this =)

**Comment by bmatcuk** [comment]
@razor-x: newer versions of brunch allow me to specify that brunch-static is outputting a "template" without causing a bunch of crap to get written to the javascript file. This _may_ solve your issue here.

Could you grab the newest version of brunch-static (v1.1.1) and brunch (v2.4.2) and see if this solves the problem without a work around?

**Comment by razor-x** [comment]
Seems like this is working now with my original test case. :cool:
